apps/root/src/common/utils/earn/parsing.tsx

```diff
@@ -63,24 +61,42 @@ export const parseAllStrategies = ({
   tokenList,
   intl,
 }: {
-  strategies: ApiStrategy[];
+  strategies: SdkStrategy[];
   tokenList: TokenList;
   intl: ReturnType<typeof useIntl>;
 }): Strategy[] =>
   strategies.map((strategy) => {
-    const network = find(NETWORKS, { chainId: strategy.chainId }) as NetworkStruct;
+    const network = find(NETWORKS, { chainId: strategy.farm.chainId }) as NetworkStruct;
+    const assetToken = sdkStrategyTokenToToken(
+      strategy.farm.asset,
+      `${strategy.farm.chainId}-${strategy.farm.asset.address}` as TokenListId,
+      tokenList
+    );
 
     return {
       id: strategy.id,
-      asset: earnApiTokenToTokenWithIcon(strategy.asset, strategy.chainId, tokenList),
-      rewards: Object.values(strategy.rewards).map((reward) => ({
-        token: earnApiTokenToTokenWithIcon(reward.token, strategy.chainId, tokenList),
-        apy: reward.apy,
-      })),
+      asset: {
+        ...assetToken,
+        icon: <TokenIcon token={assetToken} />,
```

Comment:
I'm not sure we should start sending a lot of react components through parsing, not really sold on the idea, I would try to keep the parsing from address/asset into React component as far along the component tree as we can. Since we cannot store react components for example

Reply:
tsx => tx, got it! I remember now the issue you faced with redux with a similar parsing function